### DISCUSSION TITLE
ESP-NOW support for purpose of remote control buttons

### DISCUSSION THREAD

**Discussion by goatchurchprime** [arendst/Tasmota#15388] [Ideas]
Many of the Tasmota power plugs have a little clicky button on the side so you can turn them on and off without having to go through a network app.  This is quite inconvenient if the plug socket is close to the floor behind the sofa.  It would be useful if we could make a little battery powered switch to put on the wall or the table with a button to click to operate it remotely. 

It looks like the ESP-NOW technology was designed for this purpose.  If Tasmota had this, then we could replicate this widely sold product.
 
![image](https://user-images.githubusercontent.com/677254/163713971-cba9b800-9584-4110-a4f5-ab01126d1c1e.png)

**Comment by barbudor** [comment]
You can use RF switches and remotes with a RF Bridge or ZigBee switches and remotes with a Zigbee bridge

ESP-NOW is a Wifi Mesh and it is used already in TasmotaMesh
But it doesn't solve the power consumption issue related to Wifi
This is why Zigbee is currently the best solution

Bluetooth could be too but while there are Bluetooth sensors around, I haven't seen much buttons/switches/remotes using Bluetooth

**Comment by goatchurchprime** [comment]
The [espressif page on ESP-NOW](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp_now.html) says it's designed for remote controls.  This is how I first heard of ESP-NOW, where it was used to implement [EspNowSkip](https://revspace.nl/EspNowSkip).  Why has it been chosen for a mesh network implementation when [espressif already has ESP-Wifi-Mesh](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/network/esp-wifi-mesh.html) feature?

It's a shame if we can't possibly have a remote-control press-the-on/off-button at a distance without depending on any of these extra routers or RF-bridges.

**Comment by barbudor** [comment]
May be I misunderstood and TasmotaMesh is using this other feature

**Reply by s-hadinger** [reply]
Yes. TasmotaMesh is already using esp-now so we can expect it to be relatively easy to support wiz remotes

**Comment by gustavolaureano** [comment]
Hi, adding some info to this discussion, I agree with @goatchurchprime and also think this would be a killer functionality for Tasmota

I bought some wizmotes some time ago in a sale (they were really cheap and are basically ESP NOW remotes) and now I am using it with WLED and it is as easy and convenient as I thought it would be.

It was recently added to the WLED project and works like a charm, here is the pull request, quite well documented btw:
https://github.com/Aircoookie/WLED/pull/3237
